File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        def helper(root, level, result):
            if not root:
                return 
            
            if len(result) <= level:
                result.append([])

            result[level].append(root.val)

            helper(root.left, level + 1, result)
            helper(root.right, level + 1, result)
        
        solution = []
        helper(root, 0, solution)

        return solution

File: Data Structures & Algorithms/same-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
bool same = true;
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        if (!p and !q) {
            return true;
        }
        
        if ((!p and q) or (!q and p) or p->val != q->val) {
            same = false;
            return false;
        }

        isSameTree(p->left, q->left);
        isSameTree(p->right, q->right);

        return same;
    }
};
